### PR TITLE
Use compilation mode in *helm-ag-edit* buffers

### PR DESCRIPTION
**Pull request by thomasjm**
This is a sketch of how `helm-ag` can use `compilation-minor-mode` when it's in "edit" mode. I think it's really useful--it lets you jump around to occurrences.

(As an experiment, try this on master branch without my change: create a `*helm-ag-edit*` buffer and then do `M-x compilation-minor-mode`. Compilation minor mode automatically parses the file and makes hyperlinks out of the file names.)

I believe this eliminates the need for `helm-ag--run-save-buffer`, because now you can both edit *and* navigate using the `*helm-ag-edit*` buffer.

(I'm hoping a better elisp programmer than me can figure out how to enable `compilation-minor-mode` in a less sketchy way, without screwing up fundamental-mode.)

Please take a look!

### CODE REVIEW

File: helm-ag.el [hunk 1]
```diff
@@ -403,8 +403,9 @@ They are specified to `--ignore' options."
 
 (defvar helm-ag-edit-map
   (let ((map (make-sparse-keymap)))
-    (define-key map (kbd "C-c C-c") 'helm-ag--edit-commit)
-    (define-key map (kbd "C-c C-k") 'helm-ag--edit-abort)
+    ;; Renamed to not conflict with compilation-minor-mode
+    (define-key map (kbd "C-c C-s") 'helm-ag--edit-commit)
+    (define-key map (kbd "C-c C-q") 'helm-ag--edit-abort)
     map))
```
Comment (syohex) [inline helm-ag.el]: Why do you change key maps ?

**Comment by syohex** [comment]
This breaks compatibilities.

> This is a sketch of how helm-ag can use compilation-minor-mode when it's in "edit" mode. I think it's really useful--it lets you jump around to occurrences.

We cannot edit in `compilation-minor-mode`, because there are no `self-insert-command` in its map.

I think you should use [ag.el](https://github.com/Wilfred/ag.el), it is friendly with compilation mode.

**Comment by thomasjm** [comment]
No no, you *can* edit. You should try my branch, it works! (For some reason, enabling and then disabling the mode causes the navigation functionality to still work, but I'm sure there's a better way to selectively bring over functionality from `compilation-minor-mode`.)

**Comment by syohex** [comment]
Here is screencast of your patch version. I type some key then Emacs says key is undefined.

helm-ag

**Comment by thomasjm** [comment]
Hmm, you shouldn't have the minor mode enabled in the *helm-ag-edit* buffer. I just pushed another change to explicitly disable it; verified that it works for me. Would you mind pulling my branch and giving it another try?

**Comment by syohex** [comment]
It works. But I would rather not want to change existing key bindings and remove existing interfaces.
I change code for using next-error and previous-error at #90 and merge it. Please check it.

**Pull request closed without merging**